src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -1197,14 +1199,18 @@ void Transformations::MainSnippets() {
     // shouldn't tokenize dynamic Subgraphs - it will lead to performance degradations
     bool is_dynamic_mha_token_enabled = config.snippetsCacheCapacity != 0;
 #if defined(OPENVINO_ARCH_ARM64)
-    // ARM has 32 gprs. After excluding 2 registers for work amounts, 1 register for runtime parameters, 1 platform
-    // register, 3 registers for temporary use, and 2 stack related registers, it has 23 remaining registers.
-    size_t data_ptr_gpr_count = 23;
-    snippets::pass::SnippetsTokenization::Config::CanBeFusedAsPostOpPred supported_as_postop = nullptr;
+    // ARM has 32 gprs, but the following registers should be excluded from available registers:
+    // - abi_param1: used for runtime parameters
+    // - X18: platform register
+    // - 2 (SP, X29) stack related registers
+    // - 3 (X_TMP_0, X_TMP_1, X_DEFAULT_ADDR) registers for temporary use
+    size_t available_gprs_count = 25;
```

Comment:
The point of discussion for the next PR (no need to address now): since we have an information of excluded registers, why don't we use `"total_gprs" - "excluded_regs".size()` formula.

Here we will not have extra comments that are easy to be out of date and the code will be self-documented